install.sh

```diff
@@ -3973,6 +4041,76 @@ install_into_existing_repositories() {
     return
 }
 
+############################################################
+# Install the new Git hook templates into
+#   all repos registered for autoupdate.
+#
+# Returns: 0
+############################################################
+install_into_registered_repositories() {
+
+    LIST="$INSTALL_DIR/autoupdate/registered"
+    if [ -f "$LIST" ]; then
+
+        # Filter list according to
+        # - non-existing repos
+        # - already installed
+        # - if marked as single install.
+        TEMP_LIST=$(mktemp) # temporary list for user
+        NEW_LIST=$(mktemp)  # the new list
+        IFS="$IFS_NEWLINE"
+        while read -r INSTALLED_REPO; do
+            unset IFS
+
+            if [ ! -d "$INSTALLED_REPO" ] ||
```

Comment:
Sorry, I can't really work it out from this.
What is this `[ ! -d "$INSTALLED_REPO" ] || echo "$EXISTING_REPOSITORY_LIST" | grep -q "$INSTALLED_REPO";` case?
We had it in the `autoupdate/registered` file, but the repo doesn't exist or if the user let us search for the directories and it was on it? (don't understand the meaning of this)

Also can you explain what are `$LIST`, `$NEW_LIST` and `$TEMP_LIST` please? Hopefully we can work out a better name for them.